base/src/consensus_state.rs

```diff
@@ -81,9 +86,129 @@ pub struct ConsensusState {
     pub supercharge_coinbase: bool,
 }
 
+/// For ConsensusConstants Default
+pub struct ConsensusConstants {
+    /// Point of finality (number of confirmations)
+    pub k: Length,
+    /// Number of slots per epoch
+    pub slots_per_epoch: Length,
+    /// No of slots in a sub-window = 7
+    pub slots_per_sub_window: Length,
+    /// Maximum permissable delay of packets (in slots after the current)
+    pub delta: Length,
+    /// Timestamp of genesis block in unixtime
+    pub genesis_state_timestamp: BlockTime,
+    /// Sub windows within a window
+    pub sub_windows_per_window: Length,
+    /// Number of slots before minimum density is used in chain selection
+    pub grace_period_end: Length,
+}
+
+impl Default for ConsensusConstants {
+    fn default() -> Self {
+        // TODO: read from config
+        Self {
+            k: Length(290),
+            slots_per_epoch: 7140.into(),
+            slots_per_sub_window: 7.into(),
+            delta: 0.into(),
+            genesis_state_timestamp: BlockTime(1615939200000),
+            sub_windows_per_window: 11.into(),
+            grace_period_end: Length(1440),
+        }
+    }
+}
+
 impl ConsensusState {
     /// Returns the sub-window densities as a vec of u32
     pub fn sub_window_densities(&self) -> Vec<u32> {
         self.sub_window_densities.iter().map(|i| i.0).collect()
     }
 }
+
+impl Default for ConsensusState {
+    // TODO: read from config
+    fn default() -> Self {
+        let total_currency = Amount(805385692840039233);
+        let staking_epoch_data = {
+            let mut data = EpochData::default();
+
+            data.epoch_length.0 = 1;
+            data.ledger.hash =
+                LedgerHash::from_base58("jx7buQVWFLsXTtzRgSxbYcT8EYLS8KCZbLrfDcJxMtyy4thw2Ee")
```

Comment:
same for this function, let's document where the values are from

Reply:
From our previous codebase for the new default value. Will document these

Reply:
Fixed document [Initialize consensus JSON template](https://github.com/MinaProtocol/mina/blob/32f529d8d8d712a44ee75be66061ce08cbdc8924/docs/specs/consensus/README.md#61-initialize-consensus)